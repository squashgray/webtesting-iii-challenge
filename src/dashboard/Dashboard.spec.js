import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"; // for toBeInTheDocument to work
import Dashboard from "./Dashboard";

test("should match snapshot", () => {
  const wrapper = render(<Dashboard />);
  expect(wrapper.asFragment()).toMatchSnapshot(); // elements are rendering correctly //
});

test("defaults to unlocked and open", () => {
  const { getByText } = render(<Dashboard />);
  const unlocked = getByText(/unlocked/i);
  const open = getByText(/open/i);

  expect(unlocked).toBeInTheDocument();
  expect(open).toBeInTheDocument(); //finds unlocked and open on the dashboard //
});
