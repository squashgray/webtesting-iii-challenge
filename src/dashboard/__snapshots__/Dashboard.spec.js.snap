// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="locked"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="doors"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="lock"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="door"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
